src/core/assert.shame.d.ts

```diff
@@ -0,0 +1,6 @@
+declare module '#core/assert' {
```

Comment:
Why shame?

Reply:
Shame files indicate that we're stubbing types for a module to allow the rest to pass, but the shame files should eventually be removed. Ex. in this case, assertions are complicated to convert because of a difference in how TS vs CC handle them, so rather than solve that problem, we exclude the `#core/assert` module entirely and instead provide extern type signatures so TS can handle call-sites without processing the actual assertion functions yet.